### ISSUE TITLE
Update to webpack 2

### ISSUE THREAD

**Issue by bjankord** [cerner/terra-core#209]
### Description of Issue
To move us to a more modern build tool, we should update to webpack 2. In order to migrate to webpack 2, we need to ensure all of our webpack plugins / dependencies we use today are compatible with webpack 2.

Work on this has started in this branch:
https://github.com/cerner/terra-core/tree/webpack2

### Expected Outcomes
- webpack config is updated to be compatible with webpack 2
- webpack plugins are updated to be compatible with webpack 2

**Comment by Matt-Butler** [comment]
I merged the webpack 2 branch with master today.

I am getting errors when running nightwatch tests after upgrading.

**Comment by windse7en** [comment]
1. Update to translation chunk name in webpack 2, fix in `terra-site/webpack.config.js`:
```
const webpack = require('webpack');
plugins:...
    new webpack.NamedChunksPlugin(),
```

2. Add `es6-promise` to the terra-site. `require.ensure` is based on that. Fix in `terra-site/src/index.jsx`:
```
import 'es6-promise/auto';
```

**Comment by bjankord** [comment]
Let's use [babel-polyfill](https://babeljs.io/docs/usage/polyfill/) instead of [es6-promise](https://www.npmjs.com/package/es6-promise) as it seems to be used by more of our consumers.

**Comment by bjankord** [comment]
Fixed in #270
